README.md

```diff
@@ -7,290 +7,505 @@
-- Now provides an interface for the [Authentication API](https://auth0.com/docs/auth-api).
+```php
+// Example #4
+// decode-jwt.php
+use Auth0\SDK\JWTVerifier;
+use Auth0\SDK\Exception\InvalidTokenException;
+use Auth0\SDK\Exception\CoreException;
 
->***Note:*** API V2 restrict the access to the endpoints via scopes. By default, the user token has granted certain scopes that let update the user metadata but not the root attributes nor app_metadata. To update this information and access another endpoints, you need an special token with this scopes granted. For more information about scopes, check [the API v2 docs](https://auth0.com/docs/apiv2Changes#6).
+// Do we have an ID token?
+if (empty($_GET[ 'id_token' ])) {
+    echo '<code>No `id_token` URL parameter!</code> ';
+    exit;
+}
+
+// Do we have a valid algorithm?
+if (empty($_GET[ 'token_alg' ]) || ! in_array($_GET[ 'token_alg' ], [ 'HS256', 'RS256' ])) {
+    echo '<code>Missing or invalid `token_alg` URL parameter!</code> ';
+    exit;
+}
 
-## Examples
+$config = [
+    'supported_algs' => [ $_GET[ 'token_alg' ] ],
+    'client_secret' => getenv('AUTH0_CLIENT_SECRET'),
+];
 
-Check the [basic-oauth](https://github.com/auth0/auth0-PHP/tree/master/examples/basic-oauth) example to see a quick demo on how the sdk works.
-You just need to create a `.env` file with the following information:
+if ('HS256' === $_GET[ 'token_alg' ]) {
+    $config['client_secret'] = getenv('AUTH0_CLIENT_SECRET');
+} else {
+    $config['authorized_iss'] = [ 'https://' . getenv('AUTH0_DOMAIN') . '/' ];
+}
 
+try {
+    $verifier = new JWTVerifier($config);
+    $decoded_token = $verifier->verifyAndDecode($_GET[ 'id_token' ]);
+    echo '<pre>' . print_r($decoded_token, true) . '</pre>';
+} catch (InvalidTokenException $e) {
+    echo 'Caught: InvalidTokenException - ' . $e->getMessage();
+} catch (CoreException $e) {
+    echo 'Caught: CoreException - ' . $e->getMessage();
+} catch (\Exception $e) {
+    echo 'Caught: Exception - ' . $e->getMessage();
+}
 ```
-AUTH0_CLIENT_SECRET=YOUR_APP_SECRET
-AUTH0_CLIENT_ID=YOU_APP_CLIENT
-AUTH0_DOMAIN=YOUR_DOMAIN.auth0.com
-AUTH0_CALLBACK_URL=http://localhost:3000/index.php
-AUTH0_APPTOKEN=A_VALID_APPTOKEN_WITH_CREATE_USER_SCOPE
+
+Additional parameters for the `JWTVerifier` configuration array are:
+
+- **cache**: Receives an instance of `Auth0\SDK\Helpers\Cache\CacheHandler` (Supported `FileSystemCacheHandler` and `NoCacheHandler`). Defaults to `NoCacheHandler` (RS256 only).
+- **guzzle_options**: Configuration propagated to Guzzle when fetching the JWKs (RS256 only). These options are [documented here](http://docs.guzzlephp.org/en/stable/request-options.html).
+- **secret\_base64\_encoded**: When `true`, it will decode the secret used to verify the token signature. This is only used for HS256 tokens and defaults to `true`. Your Application settings will say whether the Client Secret provided is encoded or not. 
```

Comment:
better to double check this with a CURL, but I think this no longer applies for "new tenants". AFAIK you can't PATCH a `client` and change that secret base64 encoded value. If the changes are going to be released on a new major, think of removing this.

Reply:
Application settings show whether it is or not:

![screenshot 2018-05-03 15 35 17](https://user-images.githubusercontent.com/855223/39605947-bc3ea328-4ee7-11e8-8cc6-fb4f57a037bf.png)

That's more straight-forward for docs than walking through a Management cURL.